src/main/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTask.java

```diff
@@ -65,12 +66,18 @@ public void run() {
 
                 // remove pdf content, as it's no longer needed
                 letter.setPdf(null);
+            } catch (FtpException exception) {
+                letter.setState(LetterState.FailedToUpload);
```

Comment:
so we'll never try to upload it again?

Reply:
There can be other task to loop through failed letters as it is specific scenario. If problem is continuous we assuring to not keep on hitting with same (most likely bigger) volume of documents.
This is as per suggestion in previous PR plus functionality kind of ported from producer/consumer repos